### ISSUE TITLE
log() output different for negated quaternions

### ISSUE THREAD

**Issue by lorenwel** [ethz-asl/two_state_information_filter#22]
Negated quaternions still express the same rotations. 
Currently, the quaternion logarithm is computed using the atan2 function: `2*atan2(sha,re)/sha*im`
This causes logarithm values to be different for positive and negative quaternions expressing the same rotations:
```matlab
function [ out ] = quat_log( q )
  a = q(1);
  v = q(2:end);
  out = 2*atan2(norm(v), a)*v/norm(v);
end

quat_log([-0.999 -0.002 0 0])   =  [-6.2792         0         0]
quat_log([0.999 0.002 0 0])     =  [0.0040         0         0]
```
While this is technically correct, using this in terms of rotation residuals does not make sense. We would much rather have a smaller residual in the *"wrong"* direction than having a huge residual in the *"correct"* direction. 

This can cause the estimated rotation quaternion to also negate in certain circumstances. My guess is this happens when the residual becomes larger than the small angle log() approximation, which already is symmetric around 0: `((std::signbit(re)!=0)*-2+1)*2*im`
This causes tracking issues while the quaternion is being negated: 
![image](https://user-images.githubusercontent.com/5803027/29605860-b2106ef8-87ec-11e7-97e5-6f879770ddc5.png)
Using atan instead of atan2 makes the log() function symmetric around 0 and fixes this issue: 
`2*atan(sha/re)/sha*im`
```matlab
function [ out ] = quat_log( q )
  a = q(1);
  v = q(2:end);
  out = 2*atan(norm(v)/ a)*v/norm(v);
end

quat_log([-0.999 -0.002 0 0])   =  [0.0040          0         0]
quat_log([0.999 0.002 0 0])     =  [0.0040         0         0]
```

edit: Added PR #23

**Comment by bloesch** [comment]
thanks for catching this! I checked the PR and did not notice any major effect on the currently implemented filters, but this might depend on whether orientation residuals are above the threshold for small angle approximation or not...

**Comment by lorenwel** [comment]
Closing this issue because it has been fixed with #23
